app/actions/handlers.py

```diff
@@ -0,0 +1,162 @@
+import httpx
+import logging
+
+import app.actions.client as client
+
+from datetime import datetime, timedelta, timezone
+from math import ceil
+from app.actions.configurations import AuthenticateConfig, PullObservationsConfig, PullStationHistoryConfig, get_auth_config
+from app.services.action_scheduler import trigger_action
+from app.services.activity_logger import activity_logger
+from app.services.gundi import send_observations_to_gundi
+from app.services.state import IntegrationStateManager
+from app.services.utils import generate_batches
+
+logger = logging.getLogger(__name__)
+state_manager = IntegrationStateManager()
+
+
+VW_BASE_URL = "https://www.vitalweather.co.za/api/v1"
+
+
+def transform(station, observations):
+    transformed_data = []
+    def match_units(history, units):
+        for record in history:
+            for key, value in record.items():
+                if key in units and key != "ts":
+                    record[key] = f"{value} {units[key]}"
+        return history
+
+    readings = match_units([h.dict(by_alias=True) for h in observations.History], observations.unites.dict())
+
+    for reading in readings:
+        transformed_data.append(
+            {
+                "source_name": station.Station_Name,
+                "source": station.Station_ID,
+                "type": "stationary-object",
+                "subtype": "weather_station",
+                "recorded_at": reading.pop("ts"),
+                "location": {
+                    "lat": station.latitude,
+                    "lon": station.longitude
+                },
+                "additional": {
+                    "station_height": station.height,
+                    **reading
+                }
+            }
+        )
+
+
+    return transformed_data
```

Comment:
@vgarcia13 are you sure you want to load all the records into memory before sending to Gundi?
Recall the same logic in Tahmo which caused out of memory errors. Would you consider using a generator here instead?